### PR TITLE
guymager: init at 0.8.13

### PR DESCRIPTION
**Pull request by D3vil0p3r**
To be checked in https://github.com/NixOS/nixpkgs/issues/81418

Depends on:
* https://github.com/NixOS/nixpkgs/pull/287912
* https://github.com/NixOS/nixpkgs/pull/278343

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/libguytools/default.nix
Comment (fabaff) [inline pkgs/development/libraries/libguytools/default.nix:52]: According to the [file header](https://sourceforge.net/p/libguytools/code/HEAD/tree/tags/tools-2.1.0/toolerror.cpp#l20) is the license `GPLv2+`. 

```suggestion
    license = licenses.gpl2Plus;
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gu/guymager/package.nix
Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix:53]: > This specific line goes better at the `postFixup` phase.
> 
 The line to move to `postFixup is only 53 (makeWrapper)?

File: pkgs/by-name/gu/guymager/package.nix
Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix]: Fixed.

File: pkgs/by-name/gu/guymager/package.nix
Comment (emilytrau) [inline pkgs/by-name/gu/guymager/package.nix:55]:
```suggestion
    description = "Forensic imager for media acquisition";
```

**Review by emilytrau — COMMENTED** [review]
- Needs split into a commit for each package
- Needs rebase

**Comment by D3vil0p3r** [comment]
> * Needs split into a commit for each package
> * Needs rebase

@emilytrau fixed

**Comment by sarcasticadmin** [comment]
Result of `nixpkgs-review pr 278529` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>guymager</li>
    <li>libguytools</li>
  </ul>
</details>

**Review by sarcasticadmin — APPROVED** [review]

File: pkgs/by-name/li/libguytools/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/li/libguytools/package.nix:45]:
```suggestion
    description = "Small programming toolbox";
```

File: pkgs/by-name/li/libguytools/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/li/libguytools/package.nix:44]:
```suggestion
  meta = {
```
tracking issue #292468

File: pkgs/by-name/li/libguytools/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/li/libguytools/package.nix:50]:
```suggestion
    maintainers = with lib.maintainers; [ d3vil0p3r ];
    platforms = lib.platforms.unix;
    license = lib.licenses.gpl2Plus;
```

File: pkgs/by-name/li/libguytools/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/li/libguytools/package.nix:24]: why?

File: pkgs/by-name/gu/guymager/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/gu/guymager/package.nix:61]:
```suggestion
  meta = {
    description = "Forensic imager for media acquisition";
    mainProgram = "guymager";
    homepage = "https://guymager.sourceforge.io";
    maintainers = with lib.maintainers; [ d3vil0p3r ];
    platforms = lib.platforms.unix;
    sourceProvenance = with lib.sourceTypes; [ binaryBytecode ];
    license = lib.licenses.gpl2Only;
```
might be helpful to add a comment as to what the bytecode is coming from?

File: pkgs/by-name/gu/guymager/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/gu/guymager/package.nix:9]: this doesn't exist in the nixpkgs rev you're using, please rebase on current `master`

**Review by eclairevoyant — CHANGES_REQUESTED** [review]

File: pkgs/by-name/li/libguytools/package.nix
Comment (eclairevoyant) [reply pkgs/by-name/li/libguytools/package.nix:50]: Also, is this supported on darwin? Project page only mentions linux support, and CI fails to build it on darwin

File: pkgs/by-name/li/libguytools/package.nix
Comment (D3vil0p3r) [reply pkgs/by-name/li/libguytools/package.nix:24]: @eclairevoyant If I remove that, I get the following error:
```
Running phase: qtPreHook
Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
error: builder for '/nix/store/fncpc8jyf7b1h08yipf25pkmls71pxhn-libguytools-2.1.0.drv' failed with exit code 1;
       last 10 log lines:
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/qqzz17mjs51kpfx9faw98f35r7ffwqng-xcb-util-renderutil-0.3.10-dev
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/7rd4b08axmpg5nnys8kkjbmj7cxlkdlc-xcb-util-renderutil-0.3.10
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/x3l69111mbfd38dd8q59hqv94vz6d6jn-xcb-util-wm-0.4.2-dev
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/fiwwkqgfgb4g2f63iycpsx2h5amk2jdv-xcb-util-wm-0.4.2
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/h5j6ggf0g230bavqarsvyi23zw789fv3-libglvnd-1.7.0-dev
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/d6gvgzzifggrb7fh1v0yi8bvrdlwhpqa-libglvnd-1.7.0
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/hjspqdvym6v553582mxp2a3f84p4aqyr-qtbase-5.15.14-bin
       > calling 'envTargetTargetHook' function hook 'make_glib_find_gsettings_schemas' /nix/store/9wba9diw8b199hfzcv8w99wzma66dzm1-qtbase-5.15.14
       > Running phase: qtPreHook
       > Error: wrapQtAppsHook is not used, and dontWrapQtApps is not set.
       For full logs, run 'nix log /nix/store/fncpc8jyf7b1h08yipf25pkmls71pxhn-libguytools-2.1.0.drv'.
```
I can also use only:
```
nativeBuildInputs = [ libsForQt5.qmake libsForQt5.wrapQtAppsHook ];
```
and remove `dontWrapQtApps`. What should I use?

File: pkgs/by-name/gu/guymager/package.nix
Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix:9]: Just for confirm, the right commands to do this are:
```
git clone https://github.com/D3vil0p3r/nixpkgs -b patch-9
git remote add upstream https://github.com/NixOS/nixpkgs.git
git fetch upstream
git rebase upstream/master
git push -f origin patch-9
```
?

Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix:9]: After `git rebase upstream/master`, there are a lot of files that need to fix conflicts... 
```
...
pkgs/top-level/release-lib.nix: needs merge
pkgs/top-level/release-python.nix: needs merge
pkgs/top-level/release-r.nix: needs merge
pkgs/top-level/release-small.nix: needs merge
pkgs/top-level/release.nix: needs merge
pkgs/top-level/ruby-packages.nix: needs merge
pkgs/top-level/splice.nix: needs merge
pkgs/top-level/stage.nix: needs merge
pkgs/top-level/unixtools.nix: needs merge
pkgs/top-level/wine-packages.nix: needs merge
You must edit all merge conflicts and then
mark them as resolved using git add
```
Probably it is better to close this PR and open a new one?

Comment (AndersonTorres) [reply pkgs/by-name/gu/guymager/package.nix:9]: GitHub interface is sub subpar.

You can use the bulldozer method:

- Save guymager/package.nix to a safe location (say ~/backups/)
- create a local `patch-9` branch
- get back guymager/package.nix
- commit and force-push

Comment (eclairevoyant) [reply pkgs/by-name/gu/guymager/package.nix:9]: @D3vil0p3r you should already have a local clone of nixpkgs if you're creating PRs, as rebasing is often necessary.
Ensure you _do not_ use a shallow or treeless clone; use a blobless clone if you need to save time (`--filter=blob:none`)

Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix:9]: So, if I understood well, I should blobless clone the nixpkgs repo of my `patch-9`:
```
git clone https://github.com/D3vil0p3r/nixpkgs -b patch-9 --filter=blob:none
cd nixpkgs
git rebase -i
```
but I get only `noop`. Should I do additional things? Sorry for wasting your time guys.

File: pkgs/by-name/gu/guymager/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/gu/guymager/package.nix:2]: Please format with the new formatter.

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gu/guymager/package.nix
Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix]: Fixed

File: pkgs/by-name/gu/guymager/package.nix
Comment (FliegendeWurst) [inline pkgs/by-name/gu/guymager/package.nix:80]:
```suggestion
    sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
```
Why not build from source?
https://sourceforge.net/p/guymager/code/HEAD/tree/tags/guymager-0.8.13/

Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix]: Because the date of latest commit corresponds to the date of the latest stable release. So I guess they are the same.

Comment (FliegendeWurst) [reply pkgs/by-name/gu/guymager/package.nix]: We usually try to build from the source. In this case it may be a bit difficult as there is no Makefile in the source. I suppose it's okay to use the release build then.

Comment (D3vil0p3r) [reply pkgs/by-name/gu/guymager/package.nix]: If there are no additional issues, do you have the grants to merge it?

**Review by kirillrdy — APPROVED** [review]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 278529`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>guymager</li>
    <li>libguytools</li>
  </ul>
</details>

**Comment by kirillrdy** [comment]
will merge in 24 hours unless someone comments more

File: pkgs/by-name/li/libguytools/package.nix
Comment (kirillrdy) [inline pkgs/by-name/li/libguytools/package.nix:53]: https://github.com/NixOS/nixpkgs/pull/278529/checks?check_run_id=32689131906

is darwin supported ?

Comment (D3vil0p3r) [reply pkgs/by-name/li/libguytools/package.nix:53]: I don't have an endpoint where to actually test it.

**Comment by FliegendeWurst** [comment]
@ofborg build libguytools

File: pkgs/by-name/li/libguytools/package.nix
Comment (FliegendeWurst) [reply pkgs/by-name/li/libguytools/package.nix:53]: The build failure is just due to GHC not being present in the cache. (and it doesn't build in the timeout for PRs)

Comment (D3vil0p3r) [reply pkgs/by-name/li/libguytools/package.nix:53]: How can we proceed?

Comment (FliegendeWurst) [reply pkgs/by-name/li/libguytools/package.nix:53]: Either wait until GHC is built successfully, merge anyway or find someone with a darwin builder.

Comment (kirillrdy) [reply pkgs/by-name/li/libguytools/package.nix:53]: looks like its broken on darwin anyway

```
error: builder for '/nix/store/sn4lpclhdid87q30g0dlyvbwjkr0a3ns-libguytools-2.1.0.drv' failed with exit code 2;
       last 25 log lines:
       >       (*ToolSignalLocal.pLogFn)(Error, pthread_self(), pFileName, pFunctionName, LineNr, pFormat, VaList);
       >                                        ^~~~~~~~~~~~~~
       > toolsignal.cpp:271:20: error: incompatible pointer to integer conversion assigning to 'unsigned long long' from 'pthread_t _Nonnull' (aka '_opaque_pthread_t *')
       >       *pThreadId = pthread_self();
       >                    ^~~~~~~~~~~~~~
       > 2 errors generated.
       > make: *** [Makefile:421: toolsignal.o] Error 1
       > make: *** Waiting for unfinished jobs....
       > toolsysinfo.cpp:40:10: fatal error: 'linux/if.h' file not found
       > #include <linux/if.h>
       >          ^~~~~~~~~~~~
       > toollog.cpp:116:87: error: cast from pointer to smaller type 'unsigned int' loses information
       >       wr += (size_t) snprintf (&LogLineHeader[wr], sizeof(LogLineHeader)-wr, "%08X ", (unsigned int)pthread_self());
       >                                                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > toollog.cpp:130:18: error: use of undeclared identifier 'fopen64'; did you mean 'fopen'?
       >          pFile = fopen64 (pLogFilename, "a");
       >                  ^~~~~~~
       >                  fopen
       > /nix/store/lq1j6jzsd0av6cm3yd9fxkjbw1mj2991-apple-sdk-10.14.6/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/stdio.h:153:7: note: 'fopen' declared here
       > FILE    *fopen(const char * __restrict __filename, const char * __restrict __mode) __DARWIN_ALIAS_STARTING(__MAC_10_6, __IPHONE_2_0, __DARWIN_ALIAS(fopen));
       >          ^
       > 1 error generated.
       > make: *** [Makefile:415: toolsysinfo.o] Error 1
       > 2 errors generated.
       > make: *** [Makefile:404: toollog.o] Error 1
       For full logs, run 'nix-store -l /nix/store/sn4lpclhdid87q30g0dlyvbwjkr0a3ns-libguytools-2.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/cpccgqf4yzaihq4v0bmmc98icdalyr55-review-shell.drv' failed to build

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/278529

--------- Report for 'x86_64-darwin' ---------
1 package marked as broken and skipped:
guymager

1 package failed to build:
libguytools
```

maybe easier to just mark is as broken on darwin, file an issue upstream

Comment (D3vil0p3r) [reply pkgs/by-name/li/libguytools/package.nix:53]: Issue reported to the guymager developer

**Comment by D3vil0p3r** [comment]
@kirillrdy can you check now please?

**Comment by FliegendeWurst** [comment]
You need to rebase on master, the libbfio package is not included in your branch.

**Comment by D3vil0p3r** [comment]
> You need to rebase on master, the libbfio package is not included in your branch.

Rebased to master. Let me know if it's ok.

**Pull request merged**
